include/devicetree.h

```diff
@@ -442,6 +442,39 @@
  */
 #define DT_PROP_LEN(node_id, prop) DT_PROP(node_id, prop##_LEN)
 
+/**
+ * @brief Get a property's logical length or a default value
+ *
+ * Here, "length" is a number of elements, which may differ from the
+ * property's size in bytes.
+ *
+ * The return value depends on the property's type:
+ *
+ * - for types array, string-array, and uint8-array, this expands
+ *   to the number of elements in the array
+ * - for type phandles, this expands to the number of phandles
+ * - for type phandle-array, this expands to the number of
+ *   phandle and specifier blocks in the property
+ * - for a default value if the property does not exists
+ *
+ * These properties are handled as special cases:
+ *
+ * - reg property: use DT_NUM_REGS(node_id) instead
+ * - interrupts property: use DT_NUM_IRQS(node_id) instead
+ *
+ * It is an error to use this macro with the reg or interrupts properties.
+ *
+ * For other properties, behavior is undefined.
+ *
+ * @param node_id node identifier
+ * @param prop a lowercase-and-underscores property with a logical length
+ * @param default_value a fallback value to expand to
+ * @return the property's length or the given default value
+ */
+#define DT_PROP_LEN_OR(node_id, prop, default_value) \
+	COND_CODE_1(DT_NODE_HAS_PROP(node_id, prop), \
+		    (DT_PROP(node_id, prop##_LEN)), (default_value))
```

Comment:
Use DT_PROP_LEN() instead of copy/pasting the implementation.

```
		    (DT_PROP_LEN(node_id, prop)), (default_value))
```

Reply:
My use case is 0, but this one is more generic and more similar to other APIs, that was the only reason.

Reply:
OK, no objections, but please do use DT_PROP_LEN() instead of inlining it.

Reply:
done